tasks/eva_2850/fix_discordant_variants.py

```diff
@@ -0,0 +1,329 @@
+
+    if rs_with_new_start['accession'] < variant_in_db['accession']:
+        dbsnp_cve_drop_statements.append(pymongo.DeleteOne({'_id': rs_variant['_id']}))
+        dbsnp_cve_drop_statements.append(pymongo.DeleteOne({'_id': variant_in_db['_id']}))
+        dbsnp_cve_insert_statements.append(pymongo.InsertOne(rs_with_new_start))
+
+        merge_event = create_merge_event(variant_in_db, rs_with_new_start)
+        dbsnp_cvoe_insert_statements.append(pymongo.InsertOne(merge_event))
+
+        result_drop = dbsnp_cve_collection.bulk_write(requests=dbsnp_cve_drop_statements, ordered=False)
+        logger.info('There was %s old documents dropped' % result_drop.deleted_count)
+        result_cve_insert = dbsnp_cve_collection.bulk_write(requests=dbsnp_cve_insert_statements, ordered=False)
+        logger.info('There was %s new documents inserted' % result_cve_insert.inserted_count)
+        result_cvoe_insert = dbsnp_cvoe_collection.bulk_write(requests=dbsnp_cvoe_insert_statements, ordered=False)
+        logger.info('There was %s new documents inserted' % result_cvoe_insert.inserted_count)
+
+        update_ss_with_new_rs(variant_in_db['accession'], rs_with_new_start['accession'])
+
+    else:
+        dbsnp_cve_drop_statements.append(pymongo.DeleteOne({'_id': rs_variant['_id']}))
+
+        result_drop = dbsnp_cve_collection.bulk_write(requests=dbsnp_cve_drop_statements, ordered=False)
+        logger.info('There was %s old documents dropped' % result_drop.deleted_count)
+
+        update_ss_with_new_rs(rs_with_new_start['accession'], variant_in_db['accession'])
+
+
+def create_merge_event(variant_merged, variant_retained):
+    merge_event = {
+        "eventType": "MERGED",
+        "accession": variant_merged['accession'],
+        "mergeInto": variant_retained['accession'],
+        "reason": "Merged after RS discordant correction",
+        "inactiveObjects": [variant_merged]
+    }
+
+    return merge_event
+
+
+def update_ss_with_new_rs(old_rs, new_rs):
+    dbsnp_sve_collection = mongo_source.mongo_handle[mongo_source.db_name][DBSNP_SUBMITTED_VARIANT_ENTITY]
+    eva_sve_collection = mongo_source.mongo_handle[mongo_source.db_name][EVA_SUBMITTED_VARIANT_ENTITY]
+
+    filter_query = {'rs': old_rs}
+    update_value = {'$set': {'rs': new_rs}}
+
+    dbsnp_result_updated = dbsnp_sve_collection.update_many(filter_query, update_value)
+    logger.info('There was %s documents dbsnp updated' % dbsnp_result_updated.modified_count)
+    eva_result_updated = eva_sve_collection.update_many(filter_query, update_value)
+    logger.info('There was %s documents eva updated' % eva_result_updated.modified_count)
+
+
+def get_rs_variants(mongo_source, assembly, rs_list):
+    rs_filter_criteria = {'asm': assembly, 'accession': {'$in': rs_list}}
+    rs_variants = get_variants(mongo_source, DBSNP_CLUSTERED_VARIANT_ENTITY, rs_filter_criteria, 'accession')
+    return rs_variants
+
+
+def get_ss_variants(mongo_source, assembly, rs_list):
```

Comment:
Somewhere, you should filter out the ss with `allele_match=false`, or `mapweight>0`

Reply:
done